stdlib/public/SwiftRemoteMirror/SwiftRemoteMirror.cpp

```diff
@@ -66,6 +66,7 @@ swift_reflection_createReflectionContext(void *ReaderContext,
     : [](void *){ return (uint8_t)8; };
```

Comment:
Can we just synthesize a QueryDataLayoutFunction here for the two pointer sizes?  It can just recognize those two query kinds and then fail on any future query.  That should let us get read of the GetSizeSize and GetPointerSize functions completely, since they aren't actually part of the C interface and they're totally obviated by the DataLayoutQueryFunction.

You can do something like:

```
  template <size_t WordSize>
  static int minimalDataLayoutQueryFunction(...) {
    if (kind == PointerSize || kind == SizeSize) return WordSize;
    return 0;
  }
```

and then just use `minimalDataLayoutQueryFunction<4>` / `minimalDataLayoutQueryFunction<8>` below.